### PR TITLE
Andes Message Link Button

### PR DESCRIPTION
**Pull request by lelalesi**
## Description
Describe why we need these changes in the lib.
Try to answer these questions:
- What is it about?
Este Pull Request crea una nueva variante de botón exclusivamente para el Andes Message, llamado `Link Button`
- How does it work?
El `Link Button` es, como su nombre lo indica, un botón que se ubica en el lugar que ocuparía el Primary Button, su comportamiento es similar tanto del anteriormente nombrado como al del Secondary Button, la diferencia principal reside en su parte gráfica, tanto al presentarse dentro del Andes Message como al presionarlo(no mostrando ningún cambio de color al presionarlo).
El Link Button siempre que este presente en el Andes Message será de manera solitaria, es decir, no puede estar presente si se quiere colocar un Primary Button, por lo que tampoco obviamente tampoco si quisiera colocarse un Secondary Button.
## Affected component
El componente afectado únicamente es el Andes Message.
## RFC Link
If this change was discussed on RFC please paste link here.
## Screenshots / GIFs
|AndesMessageHierarchy Quiet | AndesMessageHierarchy Loud |
| --- | --- |
|Screen Shot 2020-07-02 at 18 26 05|Screen Shot 2020-07-02 at 18 26 28|
|Screen Shot 2020-07-13 at 17 46 18|Screen Shot 2020-07-13 at 17 44 45|
|Screen Shot 2020-07-13 at 17 46 06|Screen Shot 2020-07-13 at 17 44 59|
|Screen Shot 2020-07-13 at 17 45 49|Screen Shot 2020-07-13 at 17 45 11|

### Demo Test App
TestApp Andes Message Link Button Demo

## Checks
Are you sure you followed all those steps? If so, repeat after me "I solemnly swear that ...":
   - [x] I have coded an example inside the Demo App,
   - [x] I have added proper tests,
   - [x] I have modified the Changelog.md file,
   - [x] I have updated GitHub wiki,
   - [ ] I have the explicit approval of one or more members of the UX Team,
   - [x] I have checked that this code is ObjC compliant.
   - [x] I have checked that all the new public enums conform to AndesEnumStringConvertible.
## Change type
This is easy, let us know what this code is about:
   - [ ] This code adds a new component
   - [x] This code includes improvements to an existent component
   - [ ] This code improves or modifies ONLY the Demo App.

### CODE REVIEW

**Comment by furyGorgoryAdmin** [comment]
### Informative Menu <h2>  ### 

### Use any of these tags in a pull request description or leave a comment ### 
| Tag | Description |
|---|---|
| **[canary]**| Creates a build that can only be deployed with canary strategy. | 
| **[build]** |  Creates version or build from a branch that has been tagged as "no releasable". | 
| **[name:version-name]** |Specify the name of the build. For example: [build] [name:1.2.3-name] or [canary] [name:1.2.3-name].|
| **[major]** | To automatically increase the first position of the version. For example: [build] [major] or [canary] [major].|
| **[minor]** | To automatically increase the second position of the version. For example: [build] [minor] or [canary] [minor]. |
| **[patch]** | To automatically increase the third position of the version. For example: [build] [patch] or [canary] [patch].|  

[For more information about ci-cd](http://docs.furycloud.io/?lang=esp&section=businessassurance)
[For more information about semver](https://semver.org/#summary)

File: LibraryComponents/Classes/AndesButton/Config/AndesButtonViewConfigFactory.swift [hunk 1]
```diff
@@ -14,15 +14,20 @@ internal class AndesButtonViewConfigFactory {
...
-        return provide(hierarchy: buttonHierarchy, size: buttonSize, text: text, icon: icon)
+        return provide(hierarchy: buttonHierarchy,
+                       size: buttonSize,
+                       text: text,
+                       textattributes: [:],
+                       icon: icon)
     }
 
-    static func provide(hierarchy: AndesButtonHierarchyProtocol, size: AndesButtonSizeProtocol, text: String, icon: AndesButtonIcon?) -> AndesButtonViewConfig {
+    static func provide(hierarchy: AndesButtonHierarchyProtocol, size: AndesButtonSizeProtocol, text: String, textattributes: [NSAttributedString.Key: Any] = [:], icon: AndesButtonIcon?) -> AndesButtonViewConfig {
```
Comment (lelalesi) [inline LibraryComponents/Classes/AndesButton/Config/AndesButtonViewConfigFactory.swift:24]: Cambié la firma de la funcion, la que esta por encima deberia tambien poder recibir el parametro de `textAttributes` ?

File: LibraryComponents/Classes/AndesButton/Size/AndesButtonSizeMedium.swift [hunk 2]
```diff
@@ -18,3 +18,15 @@ internal struct AndesButtonSizeMedium: AndesButtonSizeProtocol {
     public var padding: CGFloat = 12
 
 }
+
+internal struct AndesLinkButtonSizeMedium: AndesButtonSizeProtocol {
```
Comment (lelalesi) [inline LibraryComponents/Classes/AndesButton/Size/AndesButtonSizeMedium.swift:22]: Mover a un nuevo file

File: LibraryComponents/Classes/AndesMessage/AndesMessage.swift [hunk 3]
```diff
@@ -143,15 +149,32 @@ import UIKit
...
+        self.onPrimaryActionPressed = handler
+        reDrawContentViewIfNeededThenUpdate()
+    }
+
     /// Should return a view depending on which message variant is selected
     private func provideView() -> AndesMessageView {
         let config = AndesMessageViewConfigFactory.provideConfig(for: self)
 
-        if let pText = primaryActionText, !pText.isEmpty {
+        if (primaryActionText != nil && !primaryActionText!.isEmpty) != (linkActionText != nil && !linkActionText!.isEmpty) {
```
Comment (lelalesi) [inline LibraryComponents/Classes/AndesMessage/AndesMessage.swift:166]: Es una logica XOR

File: LibraryComponents/Classes/AndesMessage/AndesMessage.swift [hunk 4]
```diff
@@ -143,15 +149,32 @@ import UIKit
...
             return AndesMessageWithActionsView(withConfig: config)
         } else {
             return AndesMessageDefaultView(withConfig: config)
         }
+
```
Comment (lelalesi) [inline LibraryComponents/Classes/AndesMessage/AndesMessage.swift:171]: Borrar comments

File: LibraryComponents/Classes/AndesMessage/AndesMessage.swift (hunk 3 above)
Comment (samuelsainz) [reply LibraryComponents/Classes/AndesMessage/AndesMessage.swift:166]: Agregamos un comentario arriba? Algo tipo `// if primaryAction XOR secondaryAction`

File: LibraryComponents/Classes/AndesMessage/Config/AndesMessageViewConfigFactory.swift [hunk 5]
```diff
@@ -13,14 +13,26 @@ internal class AndesMessageViewConfigFactory {
 
         var config = AndesMessageViewConfig(backgroundColor: hierarchyIns.backgroundColor, pipeColor: hierarchyIns.pipeColor, textColor: hierarchyIns.textColor, titleText: message.title, bodyText: message.body, icon: typeIns.icon, iconBackgroundColor: hierarchyIns.accentColor, isDismissable: message.isDismissable, dismissIconColor: hierarchyIns.textColor)
 
-        if let primaryText = message.primaryActionText {
+        if let primaryText = message.primaryActionText, !primaryText.isEmpty {
             config.primaryActionConfig = AndesButtonViewConfigFactory.provide(hierarchy: hierarchyIns.primaryButtonHierarchy, size: AndesButtonSizeMedium(), text: primaryText, icon: nil)
         }
 
-        if let secondaryText = message.secondaryActionText {
+        if let secondaryText = message.secondaryActionText, !secondaryText.isEmpty {
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesMessage/Config/AndesMessageViewConfigFactory.swift:20]: 👏

File: LibraryComponents/Classes/AndesMessage/Config/AndesMessageViewConfigFactory.swift [hunk 6]
```diff
@@ -13,14 +13,26 @@ internal class AndesMessageViewConfigFactory {
...
             config.secondaryActionConfig = AndesButtonViewConfigFactory.provide(hierarchy: hierarchyIns.secondaryButtonHierarchy, size: AndesButtonSizeMedium(), text: secondaryText, icon: nil)
         }
 
+        if let linkText = message.linkActionText, !linkText.isEmpty {
+            var textAttr: [NSAttributedString.Key: Any]
+
+            textAttr = message.hierarchy == .loud ? [.underlineStyle: NSUnderlineStyle.single.rawValue] : [:]
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesMessage/Config/AndesMessageViewConfigFactory.swift:27]: Puede ser un `let` no? definido en la misma línea: 

```swift
let textAttr: [NSAttributedString.Key: Any] = message.hierarchy == .loud ? [.underlineStyle: NSUnderlineStyle.single.rawValue] : [:]
```

File: LibraryComponents/Classes/AndesButton/Config/AndesButtonViewConfigFactory.swift (hunk 1 above)
Comment (lelalesi) [reply LibraryComponents/Classes/AndesButton/Config/AndesButtonViewConfigFactory.swift:24]: Genial! ahora lo cambio

File: LibraryComponents/Classes/AndesMessage/AndesMessage.swift (hunk 3 above)
Comment (lelalesi) [reply LibraryComponents/Classes/AndesMessage/AndesMessage.swift:166]: Me parece muy bueno! ahi lo agrego :D

File: LibraryComponents/Classes/AndesButton/View/AndesButtonAbstractView.swift [hunk 7]
```diff
@@ -88,7 +88,12 @@ internal class AndesButtonAbstractView: UIControl, AndesButtonView {
 
     func setText(_ text: String) {
         let attributedString = NSMutableAttributedString.init(string: text)
+        let strRange = NSRange(location: 0, length: (text as NSString).length)
         attributedString.addAttribute(.paragraphStyle, value: AndesButtonHelper.provideParagraphStyle(), range: NSRange(location: 0, length: (text as NSString).length))
```
[the remark this answers is not in the document]
Comment (lelalesi) [reply LibraryComponents/Classes/AndesButton/View/AndesButtonAbstractView.swift:92]: Ups! se me pasó esa 😅

File: LibraryComponents/Classes/AndesMessage/Config/AndesMessageViewConfigFactory.swift (hunk 6 above)
Comment (lelalesi) [reply LibraryComponents/Classes/AndesMessage/Config/AndesMessageViewConfigFactory.swift:27]: Dale! Me parece mucho mejor

**Comment by lelalesi** [comment]
Sip ahí realicé los cambios sobre el `Changelog.md` :D

**Pull request merged**